issue: 1801987 Fix segfault with an empty feed file

pasis opened the pull request:
Socket objects are linked into a cyclic linked list. The operation
of looping the list leads to a NULL pointer dereference if the list
is empty.

Don't loop the list if number of the socket objects equals to 0.
This is a special case that happens when an empty feed file is
provided. Fall back to creating a single default socket object for
the case.

swx-jenkins2 commented:
Can one of the admins verify this patch?

igor-ivanov commented:
bot:retest

igor-ivanov commented on src/sockperf.cpp:
```diff
@@ -2974,41 +2974,41 @@ static int set_sockets_from_feedfile(const char *feedfile_name) {
     fclose(file_fd);
 
-    if (!rc) {
+    if (!rc && s_fd_num > 0) {
```
please consider returning error in rc when file is empty
probably doing
```
rc = ( (SOCKPERF_ERR_NONE == rc) && (0 == s_fd_num) ? SOCKPERF_ERR_BAD_ARGUMENT : rc);
```

pasis replied:
Done.

pasis committed:
issue: 1801987 Fix segfault with an empty feed file

Socket objects are linked into a cyclic linked list. The operation
of looping the list leads to a NULL pointer dereference if the list
is empty.

Exit with error when user provides an empty feeedfile. File which
contains only comments and empty lines is considered empty as well.

Also move the block which checks for an I/O error out of the reading loop.
The block was unreachable, because (!res) is condition when the loop
stops.

Signed-off-by: Dmytro Podgornyi <dmytrop@mellanox.com>

pasis committed:
issue: 1801987 Replace malloc() with MALLOC()

Don't mix malloc() and MALLOC() to keep the code consistent.

Signed-off-by: Dmytro Podgornyi <dmytrop@mellanox.com>

igor-ivanov reviewed (commented):
please consider /malloc/MALLOC/ in all places (I see at least two more) and update commit title with `issue: xxx` prefix

pasis commented:
> please consider /malloc/MALLOC/ in all places (I see at least two more) and update commit title with `issue: xxx` prefix

Done.

igor-ivanov reviewed (approved):
👍
